feat(python): add auto venv activation

rcrews opened the pull request:
## Standards checklist:

<!-- Fill with an x the ones that apply. Example: [x] -->

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

- Adds and runs a function that listens for directory changes. When cd'ing into a directory
  containing `venv/bin/activate`, it automatically activates the virtual environment.
  When cd'ing out of the directory, it automatically deactivates the virtual environment.
- It supports two global environment variables,
  - `VENV_NAME`  where a name different from `venv` can be specified. (For example,
    `.venv` is also suggested by the Python docs.)
  - Also, if anyone doesn't want this, they can simply set `DISABLE_AUTO_VRUN` and the listener
    will always exit without making any changes.

## Other comments:

Auto-activating the environment is a feature of the Ruby Version Manager (rvm). I really like that feature,
and researched till I could add it to my Python environments. I've been running this on my computer for
about six months without issues.

rcrews commented:
Providing at-mentions to people who have contributed to the `python` plug-in in the last year: @gautamkrishnar @MikuHatsune39 @carlosala

carlosala reviewed (changes requested):
I like it! Could you make it opt-in instead of out-out? Thanks!

rcrews commented:
Thanks for your kind reaction, @carlosala! I added two commits. One does as you requested and makes the new feature opt-in, and improves the description of the new feature in the README.
Since the new feature introduces a couple new configuration environment variables, I updated the two existing venv functions -- `vrun` and `mkv` -- to also use `PYTHON_VENV_NAME`, if it is set.
Let me know if you want me to squash these into a single commit, or if you want me to add "feat(python):" to the commit messages for all of the commits.

carlosala reviewed (approved):
LGTM! 🚀

carlosala commented:
I made some changes to simplify the code and align it with our usual design.
Thanks for your contribution!

robbyrussell commented:
Hey @rcrews  -- we really appreciate the contribution! ✌️
